src/components/NavigationBar/NavigationBar.tsx

```diff
@@ -0,0 +1,335 @@
+}
+
+interface LabelButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.LabelButton;
+}
+
+interface IconButton extends Omit<NavigationLink, "schema"> {
+  schema: NavigationItemType.IconButton;
+  icon: Icon;
+}
+
+export type NavigationItem = NavigationLink | DropDownList | LabelButton | IconButton;
+
+export interface NavigationBarProps {
+  navigationItems: NavigationItem[];
+}
+
+const NavigationBarStyle = styled.nav`
+  button {
+    color: #6e787f;
+    font-size: 16px;
+    font-family: "Roboto";
+  }
+
+  a {
+    color: #6e787f;
+    display: block;
+    font-size: 16px;
+    font-family: "Roboto";
+  }
+
+  .navbar-toggler {
+    padding: 0.25rem 0.75rem;
+    background-color: transparent;
+    border: 1px solid transparent;
+    border-radius: 0.25rem;
+
+    &:hover,
+    &:focus {
+      text-decoration: none;
+      outline: none;
+    }
+
+    .icon-bar {
+      background-color: #454b50;
+      display: block;
+      width: 22px;
+      height: 2px;
+      border-radius: 1px;
+      transition: all 0.2s;
+    }
+
+    .icon-bar + .icon-bar {
+      margin-top: 10px;
+    }
+
+    .top-bar {
+      transform: rotate(45deg);
+      transform-origin: 10% 10%;
+    }
+
+    .bottom-bar {
+      transform: rotate(-45deg);
+      transform-origin: 10% 90%;
+    }
+
+    &.collapsed .top-bar {
+      transform: rotate(0);
+    }
+
+    &.collapsed .bottom-bar {
+      transform: rotate(0);
+    }
+  }
+
+  .dropdown-item {
+    font-size: 14px;
+  }
+`;
+
+export const NavigationBar: FunctionComponent<NavigationBarProps> = (props) => {
+  const [isOn, setIsOn] = useState(false);
+  return (
+    <NavigationBarStyle>
+      <div className={`container py-2`}>
+        <div className="relative flex items-center justify-between h-16">
+          <div className="absolute inset-y-0 left-0 flex items-center lg:hidden">
+            <button
+              className={`navbar-toggler outline-none ${isOn ? "" : "collapsed"}`}
+              onClick={() => {
+                setIsOn(!isOn);
+              }}
+            >
+              <span className="icon-bar top-bar" />
+              <span className="icon-bar bottom-bar" />
+            </button>
+          </div>
+          <div className="flex-1 flex items-center justify-center lg:items-stretch lg:justify-start">
+            <div className="flex-shrink-0">
+              <a href="https://www.tradetrust.io/">
+                <img
+                  data-testid="nav-logo-home"
+                  className="img-fluid h-12"
+                  src="https://www.tradetrust.io/static/images/tradetrust_logo.svg"
+                  alt="TradeTrust"
+                />
+              </a>
+            </div>
+            <div className="hidden lg:block md:ml-12">
+              <div className="flex h-full items-center">
+                {props.navigationItems.map((item, index) => {
+                  if (item.position == "left") {
+                    return (
```

Comment:
why `left`, `center`, `right`? would it be better to have 2 array of nav items instead?

Reply:
If split the nav items into two arrays may increase the complexity of the system. As I will also need to increase the number of argument passed into the functional component.